x/stakeibc/abci.go

```diff
@@ -20,7 +22,24 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper, bk types.BankKeeper, ak type
 	for _, hz := range k.GetAllHostZone(ctx) {
 		rrSafe, err := k.IsRedemptionRateWithinSafetyBounds(ctx, hz)
 		if !rrSafe {
-			panic(fmt.Sprintf("[INVARIANT BROKEN!!!] %s's RR is %s. ERR: %v", hz.GetChainId(), hz.RedemptionRate.String(), err.Error()))
+			hz.Halted = true
+			k.SetHostZone(ctx, hz)
+
+			// set rate limit on stAsset
+			stDenom := types.StAssetDenomFromHostZoneDenom(hz.HostDenom)
+			k.RatelimitKeeper.SetRateLimit(ctx, ratelimittypes.RateLimit{
+				Path: &ratelimittypes.Path{
+					Denom:     stDenom,
+					ChannelId: "", // all channel
```

Comment:
just adding a comment here for us to remember to revisit this after the changes have been made to the ratelimit module